### PR TITLE
TGBOT-45-create-object-provider-for-order-objects

### PR DESCRIPTION
**Pull request by schneidervs**
https://schneidervs.atlassian.net/browse/TGBOT-45

### CODE REVIEW

File: stoservice/src/main/java/com/tutrit/stoservice/repository/OrderRepository.java [hunk 1]
```diff
@@ -12,12 +12,13 @@
 public class OrderRepository {
 
     private static final Logger logger = LoggerFactory.getLogger(OrderRepository.class);
-    private static final Map<String, Order> orders = new HashMap();
+    public final Map<String, Order> orders = new HashMap<>();
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/OrderRepository.java:15]: why did you remove the static field?

**Review by yurasava — APPROVED** [review]

File: stoservice/src/main/java/com/tutrit/stoservice/repository/OrderRepository.java (hunk 1 above)
Comment (schneidervs) [reply stoservice/src/main/java/com/tutrit/stoservice/repository/OrderRepository.java:15]: Done

Comment (schneidervs) [reply stoservice/src/main/java/com/tutrit/stoservice/repository/OrderRepository.java:15]: Ok. WIP

File: stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    public static Order getInstanceOrder(Order order, int i) {
+        order.setId("id" + i);
+        order.setUser("UserName");
+        order.setCar("UserCar");
+        order.setOrderStatus("Open");
+        return order;
+    }
+
+    static OrderRepository createTenInstanceOrders() {
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java:18]: I think that would be covered on the scope of another issue, so could be deleted for now

File: stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+public class OrderProvider {
+    private OrderProvider() {
+    }
+
+    public static Order getInstanceOrder(Order order, int i) {
+        order.setId("id" + i);
+        order.setUser("UserName");
+        order.setCar("UserCar");
+        order.setOrderStatus("Open");
+        return order;
```
Comment (max-java) [inline stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java:15]: Either create new Order to return, or return void.

**Review by max-java — DISMISSED** [review]

File: stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+public class OrderProvider {
+    private OrderProvider(){
+        throw new AssertionError("Suppress default constructor for noninstantiability");
+    }
+
+    static Map<Object, Object> getOrders() {
+        Map<Object, Object> initialOrders = new HashMap<>();
+        for (int i = 0; i < 10; i++) {
+            initialOrders.put(getInstance(i).getId(), getInstance(i));
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java:16]: Two different objects are created here.
You can change 'getInstance(i).getId()' to '("id" + i)' to avoid creating unnecessary objects.

File: stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        }
+        return initialOrders;
+    }
+
+    static Map<Object, Object> getOrders(int amount) {
+        Map<Object, Object> initialOrders = new HashMap<>();
+        for (int i = 0; i < amount; i++) {
+            initialOrders.put(getInstance(i).getId(), getInstance(i));
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/provider/OrderProvider.java:24]: There are also two different objects here

**Review by yurasava — DISMISSED** [review]

**Review by yurasava — APPROVED** [review]
lgfm

**Review by Sav4ykVladzislau — APPROVED** [review]
lgfm

**Review by AlexanderYarokhovich — APPROVED** [review]
lgfm
